lib/api/model/model.dart

```diff
@@ -490,6 +493,12 @@ sealed class Message {
   @JsonKey(name: 'match_subject')
   final String? matchTopic;
 
+  static MessageEditState _messageEditStateFromJson(dynamic json) {
+    // The value passed here must be a MessageEditState already due to
+    // processing work done in [MessageEditState.readFromMessage].
+    return json as MessageEditState;
```

Comment:
Then we will need to have `readValue` return a string. It should work but then we don't have the type checking benefits of using the enum.

Reply:
Updated the PR so we return strings for the enums.

Reply:
Oh, I see. Looking at the [`JsonKey.readValue` docs](https://pub.dev/documentation/json_annotation/4.9.0/json_annotation/JsonKey/readValue.html), it says:
> Note: using this feature does not change any of the subsequent decoding logic for the field. For instance, if the field is of type [DateTime](https://api.dart.dev/stable/3.4.4/dart-core/DateTime-class.html) we expect the function provided here to return a [String](https://api.dart.dev/stable/3.4.4/dart-core/String-class.html).

I do like having the function return the enum values, rather than strings. (Especially as these are strings we made up — they're not even in the API.) So let's switch that back.

I think what confused me is that this `_messageEditStateFromJson` method wasn't doing any conversion, so it seemed redundant. Now I think I understand: the point is that by default the generated `fromJson` expects to do a conversion at this step, and this is here in order to avoid doing any conversion… effectively so that the `readValue` can include that conversion as part of its job.

I can try to revise the comment to something that would have successfully conveyed this for me, after you revert to add this back.

Reply:
Got it. Reverted.

Reply:
```suggestion
  static MessageEditState _messageEditStateFromJson(dynamic json) {
    // This is a no-op so that [MessageEditState._readFromMessage]
    // can return the enum value directly.
    return json as MessageEditState;
```